source/dagger/gameplay/ping_pong/player_scores.cpp

```diff
@@ -45,6 +46,39 @@ void PlayerScoresSystem::Run()
         {
             ballOnField++;
         }
+
+    }
+
+    for (auto entity : view) 
+    {
+        auto &ball = view.get<PingPongBall>(entity);
+
+        if (m_goalsPlayerOne == 5 || m_goalsPlayerTwo == 5) 
+        {
+            if (ball.processed)
+            {
+                Engine::Registry().destroy(entity);
```

Comment:
it's a bad practice to destroy objects in a Run method. Check out suggestions about how to fix it in other pull request:
https://github.com/UbiBelETF/dagger/pull/40#discussion_r522999212

Reply:
Ok, thanks for the advice!